cuML/src/pca/pca.h

```diff
@@ -77,14 +80,11 @@ void truncCompExpVars(math_t *in, math_t *components, math_t *explained_var,
  * @output param mu: mean of all the features (all the columns in the data). Size n_cols * 1.
  * @output param noise_vars: variance of the noise. Size 1 * 1 (scalar).
  * @input param prms: data structure that includes all the parameters from input size to algorithm.
- * @input param cublas_handle: cublas handle
- * @input param cusolver_handle: cusolver handle
  */
 template<typename math_t>
-void pcaFit(math_t *input, math_t *components, math_t *explained_var,
+void pcaFit(const cumlHandle_impl& handle, math_t *input, math_t *components, math_t *explained_var,
 		math_t *explained_var_ratio, math_t *singular_vals, math_t *mu,
-		math_t *noise_vars, paramsPCA prms, cublasHandle_t cublas_handle,
-		cusolverDnHandle_t cusolver_handle) {
+		math_t *noise_vars, paramsPCA prms) {
     ///@todo: make this to be passed via the interface
     DeviceAllocator mgr = makeDefaultAllocator();
```

Comment:
IMHO this PR should handle picking up `cumlHandle` fully for PCA. Therefore the allocator from `cumlHandle_impl` should be used.
I know that on PR #331 I asked to minimize the size of the PR, which might seem counter intuitive to now asking to include more. However this is exactly why i asked to handle this one algorithm at a time. So that we have the focus to do this right one after the other. Makes sense?

Reply:
@jirikraus That is just not possible to do *only* with PCA! I tried to replace the whole of PCA with all components of cumlHandle, that ended us up with PR #331 :)

In short, if we want to show full cumlHandle related changes using PCA (for example), then we'll have to accept the whole of PR #331 OR, if we just want to update the interface of one algo with corresponding cython-related proposals, then we'll need go with this PR.